gradoop-flink/src/main/java/org/gradoop/flink/algorithms/gelly/connectedcomponents/ConnectedComponents.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright © 2014 - 2017 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.flink.algorithms.gelly.connectedcomponents;
+
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.graph.Graph;
+import org.apache.flink.types.NullValue;
+import org.gradoop.common.model.impl.id.GradoopId;
+import org.gradoop.common.model.impl.pojo.Vertex;
+import org.gradoop.flink.algorithms.gelly.GellyAlgorithm;
+import org.gradoop.flink.algorithms.gelly.connectedcomponents.functions.ConnectedComponentToAttribute;
+import org.gradoop.flink.algorithms.gelly.functions.EdgeToGellyEdgeWithNullValue;
+import org.gradoop.flink.algorithms.gelly.functions.VertexToGellyVertexWithGradoopId;
+import org.gradoop.flink.model.api.epgm.LogicalGraph;
+import org.gradoop.flink.model.impl.functions.epgm.Id;
+
+/**
+ * A gradoop operator wrapping {@link org.apache.flink.graph.library.ConnectedComponents}.
+ * The result will be the same {@link LogicalGraph} with a component id assigned to each vertex
+ * as a Property.
+ */
+public class ConnectedComponents extends GellyAlgorithm<GradoopId, NullValue> {
```

Comment:
Suggestion: Rename `ConnectedComponents` into `AnnotateConnectedComponents` or something and create a dedicated class extending `UnaryGraphToCollectionOperator` which uses this ACC, `splitBy` and `map` idiom (see https://github.com/dbs-leipzig/gradoop/pull/692#issuecomment-334177400) to create `GraphCollection`. I strongly suppose, that users prefer having separate graphs than just annotations.

Reply:
To be honest, we calculate Weakly Connected Components. 
But I think `AnnotateWeaklyConnectedComponents` seems a bit long...

Reply:
Ok, so in the Gradoop Teammeeting we decide to name it that way, its fine and exact: Please rename to class to: AnnotateWeaklyConnectedComponents

Probably we want to later on add variants of the algorithm returning a GraphCollection containing the components or something similar.